articles/tutorials/qft-tutorial.md

```diff
@@ -0,0 +1,624 @@
+
+
+So, the printed output is illustrating that our programmed gates transformed our state from
+
+$$
+\ket{\psi}\_{initial} = \ket{000}
+$$
+
+to 
+
+$$
+\begin{align}
+	\ket{\psi}\_{final} &= \frac{1}{\sqrt{8}} \left( \ket{000} + \ket{001} + \ket{010} + \ket{011} + \ket{100} + \ket{101} + \ket{110} + \ket{111}  \right) \\\\
+	&= \frac{1}{\sqrt{2^n}}\sum\_{j=0}^{2^n-1} \ket{j},
+\end{align}
+$$
+
+which is precisely the behavior of the 3-qubit Fourier transform. 
+
+If you are curious about how other input states are affected, we encourage you to play around with applying qubit operations before the transform.
+
+## Adding measurements
+
+Unfortunately, a cornerstone of quantum mechanics tells us that a real quantum system cannot have such a `DumpMachine` function. 
+Instead, we're forced to extract information through measurements, which in general not only fail to provide us the full quantum state, but can also drastically alter the system itself.
+There are many sorts of quantum measurements, but we will focus on the most basic: projective measurements on single qubits.
+Upon measurement in a given basis (e.g. the computational basis $ \{ \ket{0}, \ket{1} \} $), the qubit state is projected onto whichever basis state was measured---hence destroying any superposition between the two.
+
+To implement measurements within a Q# program, we use the `M` operation (from `Microsoft.Quantum.Intrinsic`), which returns a `Result` type.
+
+First, we modify our `Perform3QubitQFT` operation to return an array of measurement results, `Result[]`, instead of `Unit`.
+
+```qsharp
+    operation Perform3QubitQFT() : Result[] {
+```
+
+#### Define and initialize `Result[]` array
+
+Before even allocating qubits (i.e. before the `using` statement), we declare and bind this length-3 array (one `Result` for each qubit): 
+
+```qsharp
+        mutable resultArray = new Result[3];
+```
+
+The `mutable` keyword prefacing `resultArray` allows the variable to be rebound later in the code---for example, when adding our measurement results.
+
+#### Perform measurements in a `for` loop and add results to array
+
+After the Fourier transorm operations inside the `using` block, insert the following code:
+
+```qsharp
+            for(i in IndexRange(qs)) {
+				set resultArray w/= i <- M(qs[i]);
+            }
+```
+The [`IndexRange`](xref:microsoft.quantum.arrays.indexrange) function called on an array (e.g. our array of qubits, `qs`) returns a range over the indices of the array. 
+Here, we use it in our `for` loop to sequentially measure each qubit using the `M(qs[i])` statement.
+Each measured `Result` type (either `Zero` or `One`) is then added to the corresponding index position in `resultArray` with an update-and-reassign statement.
+
+> [!NOTE]
+> The syntax of this statement is unique to Q#, but corresponds to the similar `set r[i] <- M(qs[i]);` seen in other languages.
```

Comment:
Other languages such as...? In my experience, traditional languages generally just say `r[i] = M(qs[i]);`. I'm not aware of any other languages that use a keyword `set` _nor_ a left-arrow `<-` for assignment, let alone both at once.

I recommend removing the note and adding a hyperlink to "Update-and-reassign statement" instead.

Reply:
F# is the most notable use of the left-arrow (also probably the inspiration for using it in Q#), but I believe R also uses it. The main difference in F# is that it uses the keyword `let` with mutable variables, whereas in Q# `set` was introduced for that purpose, with `let` is reserved for immutables. 

Thanks for the heads up though, you're absolutely right that it was unnecessarily ambiguous. Removing `set` from the example statement, and clarifying the other languages. Won't hyperlink to update-and-reassign just yet though, as that page is also undergoing a significant reorganization.